django_countries_hdx/__init__.py

```diff
@@ -7,261 +7,281 @@
+    :return: str
+    """
+    country_data = get_country_data(country.code)
 
-    def get_income_level(self, country) -> str | None:
-        """Returns whether a country's income level (from World Bank)
+    if country_data:
+        income_level = country_data.get("#indicator+incomelevel", None)
+        return income_level if income_level != "" else None
+    return None
 
-        :param country: django-countries Country object.
-        :return: str
-        """
-        country_data = self.get_country_data(country.code)
+def get_region_name(region_code: int) -> str | None:
+    """Retrieves region or sub-region name for a region code.
 
-        if country_data:
-            income_level = country_data.get("#indicator+incomelevel", None)
-            return income_level if income_level != "" else None
+    :param region_code: UN M49 region code.
+    :return: String. Region name
+    """
+    if not region_code:
         return None
 
-    def get_region_name(self, region_code: int) -> str | None:
-        """Retrieves region or sub-region name for a region code.
-
-        :param region_code: UN M49 region code.
-        :return: String. Region name
-        """
-        if not region_code:
-            return None
+    try:
+        countriesdata = HDX.countriesdata()
+        return countriesdata["regioncodes2names"].get(region_code) # noqa
+    except KeyError:
+        return None
 
-        try:
-            countriesdata = HDX.countriesdata()
-            return countriesdata["regioncodes2names"].get(region_code) # noqa
-        except KeyError:
-            return None
+def get_countries_by_region() -> dict[int, dict[str, list[tuple[str, str]]]]:
+    """Retrieves lists of countries keyed by region, with region name and country tuples.
 
-    def get_countries_by_region(self) -> dict[int, dict[str, list[tuple[str, str]]]]:
-        """Retrieves lists of countries keyed by region, with region name and country tuples.
-
-        :param regions: Boolean to determine whether to use regions or subregions.
-        :return: Dict. Keyed by region code, the value is a dictionary containing the
-        region name and a list of country_code, country_name tuples.
-        """
-        region_codes = (2, 9, 19, 142, 150,)
-
-        return {
-            region_code: {
-                "name": self.get_region_name(region_code),
-                "countries": sorted(
-                    [
-                        (
-                            HDX.get_iso2_from_iso3(code),
-                            HDX.get_country_name_from_iso3(code)
-                        )
-                        for code in HDX.get_countries_in_region(region_code)
-                    ],
-                    key=lambda x: x[1]  # Sort by country name
-                )
-            }
-            for region_code in sorted(region_codes, key=lambda x: self.get_region_name(x))
+    :param regions: Boolean to determine whether to use regions or subregions.
```

Comment:
nit: no params?

Reply:
I've added better typing for the return value as well.